server/src/service/service.rs

```diff
@@ -0,0 +1,210 @@
+//  org.freedesOnceCellktop.Secret.Service
+
+use std::collections::HashMap;
+
+use oo7::dbus::api::properties::Properties;
+use oo7::portal::Item;
+use serde::Serialize;
+use zbus::{dbus_interface, fdo, zvariant, Error, ObjectServer, SignalContext};
+use zvariant::{ObjectPath, OwnedObjectPath, OwnedValue, Type, Value};
+
+use crate::service::collection::Collection;
+use crate::service::session::Session;
+use crate::KEYRING;
+
+#[derive(Serialize, Type, Debug)]
+pub struct Service {
+    pub(self) collections: Vec<Collection>,
+}
+
+#[dbus_interface(name = "org.freedesktop.Secret.Service")]
+impl Service {
+    pub async fn open_session(
+        &self,
+        /*algorithm: &str,*/
+        input: Value<'_>,
+    ) -> fdo::Result<(OwnedValue, OwnedObjectPath)> {
+        // WIP: not complete
+        let session = Session::new(ObjectPath::try_from("/s1").unwrap().into()).await;
+        Ok((
+            OwnedValue::try_from(input).unwrap(), // avoid using unwrap
+            ObjectPath::try_from(session.path().await).unwrap().into(),
+        ))
+    }
+
+    #[dbus_interface(out_args("collection", "prompt"))]
+    pub async fn create_collection(
+        &self,
+        properties: Properties,
+        alias: String,
+        #[zbus(object_server)] object_server: &ObjectServer,
+    ) -> fdo::Result<(ObjectPath, ObjectPath)> {
+        let _attributes = properties.attributes(); // expand and pass these to Collection:new
+        let collection = Collection::new(String::from("temp")).await; // temporarily
+        let path = format!("/org/freedesktop/secrets_/collection/{}", &alias);
+
+        let _ = object_server.at(path.clone(), collection).await;
+        let new_collection_path = ObjectPath::try_from(path).unwrap(); // temporarily
+        let prompt = ObjectPath::default().into(); // temp Prompt
```

Comment:
I would suggest already creating a basic struct called Prompt, make it serialize as an ObjectPath and return it here. This way you can do `Prompt::default`, it would simplify finding out places where you still need to handle prompt later